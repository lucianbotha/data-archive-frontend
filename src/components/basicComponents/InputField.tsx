import * as React from "react";
import styled from "styled-components";

interface IInputProps extends React.InputHTMLAttributes<HTMLInputElement> {
  error?: string | undefined;
}

/**
 * A paragraph with an error message.
 */
const Error = styled.p.attrs({
  className: "has-text-danger"
})``;

/**
 * An input field with an error message underneath it.
 *
 * The field accepts all attributes you can pass to an HTML input element. In
 * addition, you may pass an error argument with an error message to display.
 */
const InputField = (props: IInputProps) => {
  const { className, error, ...rest } = props;
  return (
    <div className="control">
      <input
        type={"text"}
        className={`input ${className} ${error && error !== "" && "is-danger"}`}
        {...rest}
      />
      {error && <Error>{error}</Error>}
    </div>
  );
};
export default InputField;
