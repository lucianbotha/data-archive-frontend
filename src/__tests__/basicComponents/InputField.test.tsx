import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import * as React from "react";
import InputField from "../../components/basicComponents/InputField";

describe("InpuField", () => {
  const wrapper = shallow(<InputField />);
  it("should render", () => {
    expect(wrapper).toBeDefined();
  });

  it("always renders a div", () => {
    const divs = wrapper.find("div");
    expect(divs.length).toBeGreaterThan(0);
  });
  it("always contains an input tag with of type text", () => {
    const inputs = wrapper.find("input");
    expect(inputs.length).toEqual(1);
    const input = wrapper.find("input").get(0);
    expect(input.props.type).toEqual("text");
  });
  it("render correctly", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
    expect(
      toJson(shallow(<InputField error={"it has error"} />))
    ).toMatchSnapshot();
    expect(
      toJson(
        shallow(<InputField name="name" value="value" placeholder="holder" />)
      )
    ).toMatchSnapshot();
  });
});
