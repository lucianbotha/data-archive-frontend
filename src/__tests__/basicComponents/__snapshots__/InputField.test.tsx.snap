// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InpuField render correctly 1`] = `
<div
  className="control"
>
  <input
    className="is-label input  undefined"
    id="undefined"
    type="text"
  />
  <p />
</div>
`;

exports[`InpuField render correctly 2`] = `
<div
  className="control"
>
  <input
    className="is-label input  is-danger"
    id="undefined"
    title="it has error"
    type="text"
  />
  <p>
    it has error
  </p>
</div>
`;

exports[`InpuField render correctly 3`] = `
<div
  className="control"
>
  <input
    className="is-label input  undefined"
    id="name"
    name="name"
    placeholder="holder"
    type="text"
    value="value"
  />
  <p />
</div>
`;
